### ISSUE TITLE
C# extensions missing from GraphQLEngine

### ISSUE THREAD

**Issue by MaximS** [graphql-dotnet/conventions#141]
There is no way to add c# extensions to `GraphQLEngine` (Adapters/Engine/GraphQLEngine.cs) like it's possible for `RequestHandler` using `WithQueryExtensions`.

Please consider updating DataLoaderWithEfCore to use an extension. For example like this:

```
// GraphApi/Schema/SchemaExtensions.cs
using System;
using System.Threading.Tasks;
using AutoMapper;
using DataLoaderWithEFCore.Data.Repositories;
using GraphQL.Conventions;

namespace DataLoaderWithEFCore.GraphApi.Schema
{
    public static class MovieExt
    {
        public static string TitleWithGenre(this Movie movie)
        {
            return movie.Title + " (" + movie.Genre + ")";
        }
    }
}
```

and 

```
// GraphApi/GraphController.cs (around line 34)           
     ExecutionResult result = await _engine 
                .WithQueryExtensions(typeof(MovieExt))
                .NewExecutor()
                .WithUserContext(_userContext)
                .WithDependencyInjector(_injector)
                .WithRequest(requestBody)
                .Execute();

```

Simple adding 

        public GraphQLEngine WithQueryExtensions(System.Type typeExtensions)
        {
            _typeResolver.AddExtensions(typeExtensions);
            return this;
        }

to Adapters/Engine/GraphQLEngine.cs doesn't seem to work.

**Comment by MaximS** [comment]
Actually another `.WithQueryExtensions(typeof(MovieExt))` is required in Startup.css:

```
        public void ConfigureServices(IServiceCollection services)
        {
            services.AddMvc().SetCompatibilityVersion(CompatibilityVersion.Version_2_1);
            ....
            services.AddSingleton(provider => new GraphQLEngine()
                .WithQueryExtensions(typeof(MovieExt))
                .WithFieldResolutionStrategy(FieldResolutionStrategy.Normal)
                .BuildSchema(typeof(SchemaDefinition<Schema.Query, Schema.Mutation>)));
```
            
Please, add ` public GraphQLEngine WithQueryExtensions(System.Type typeExtensions)` to `Adapters/Engine/GraphQLEngine.cs` and may be update the docs and the `DataLoaderWithEfCore` demo.

**Comment by tlil** [comment]
Feel free to raise a PR with the proposed changes. ;-)

**Comment by tlil** [comment]
Fixed in 2.0.4
